modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache

```diff
@@ -24,7 +24,7 @@ import {
 {{#allParams.0}}
 export interface {{#prefixParameterInterfaces}}{{classname}}{{/prefixParameterInterfaces}}{{operationIdCamelCase}}Request {
 {{#allParams}}
-    {{paramName}}{{^required}}?{{/required}}: {{#isEnum}}{{{datatypeWithEnum}}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{#isNullable}} | null{{/isNullable}}{{/isEnum}};
+    {{paramName}}{{^required}}?{{/required}}: {{#isEnum}}{{{datatypeWithEnum}}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{#isNullable}}{{#required}} | null{{/required}}{{/isNullable}}{{/isEnum}};
```

Comment:
quick question: if a parameter is both optional and nullable, shouldn't it also have ` | null` as well?

Reply:
Thing is, if it's optional then the property already has a `?`, so why is ` | null` needed? It'll be the default anyway, so why restrict it?

Additionally, and specifically in TS/JS, restricting the property to only `null` means that it won't be able to receive the `undefined` value too, which in the case of JSON, ultimately resolves to `null` in transport.

```typescript

export interface SomeInterface {
  requiredNullableProperty: string | null,
  requiredProperty: string,
  nullableProperty?: string,
}
```

The last line in the example covers the situation when you may pass in a value that is either `null`, `undefined`, or indeed a `string`. If you restrict it only to `string | null`, then `undefined` won't be a plausible value, which it should.

Reply:
thanks for the explanation.

do you mind updating the PR description to include a list of these changes that are not `Use exists() to check for properties` ?

it will definitely make the review a lot easier.

Reply:
✅

Reply:
But your model is not the same as `?: string | null` - why not type in that way that omitting the property as well as setting the `null` value is allowed?

```
export interface SomeInterface {
  requiredNullableProperty: string | null,
  requiredProperty: string,
  nullableProperty?: string | null,
}
```

Reply:
Because it's pointless? :)
The extra characters have no additional value and the interpreter will see it as `nullableProperty?: string`.

Reply:
> Thing is, if it's optional then the property already has a ?, so why is  | null needed? It'll be the default anyway, so why restrict it?

This is wrong. This makes us lose the distinction between `undefined` and `nullable`. In some cases you need both. See my comment clarifying this:
https://github.com/OpenAPITools/openapi-generator/pull/17798#issuecomment-2093268320

I will raise a PR to revert this change as it broke builds for several people. Any version after 7.4.0 is now unusable because of this change.

Reply:
@l0gicgate maybe you can raise a PR to change the relevant line in `modelGenericInterfaces.mustache`, making it look like

`{{#isReadOnly}}readonly {{/isReadOnly}}{{name}}{{^required}}?{{/required}}: {{#isEnum}}{{{datatypeWithEnum}}}{{/isEnum}}{{^isEnum}}{{{datatype}}}{{/isEnum}}{{#isNullable}} | null{{/isNullable}};`

Reply:
@hendrikpeilke I did already and it's been merged as of this morning:
https://github.com/OpenAPITools/openapi-generator/pull/18887